### ISSUE TITLE
Low space error

### ISSUE THREAD

**Issue by ZoideNFT** [bitcoin-sv/bitcoin-sv#286]
This issue tracker is only for technical issues related to bitcoin-sv.

General bitcoin questions and/or support requests and are best directed to the [Bitcoin StackExchange](https://bitcoin.stackexchange.com).

For reporting security issues, please contact people privately.

### Describe the issue
Hi guys, trying to upgrade from 11 to 13 stranger things happened.  I shutted down  the node properly, changed the source folder and started it again. 

It shows "[loadblk] Error: Error: Disk space is low!" and autostop after a correct initial "last 7 blocks check out". I thought that maybe was a problem with the version 13 but, I revert the upgrade (changed folder from 13 to 11 again) but it shows the same error every time I try to start it.

I don't know why this is happening because bitcoin-data lives in a NFS folder with 8TB capacity and now is 3,5TB used. The df -h command shows a 36% usage for the root partition. 

It was working good until I upgraded. Same command at startup. Nothing changed except the folder with the new version.

### Screenshots.
If the issue is related to the GUI, screenshots can be added to this issue via drag & drop.

![image](https://user-images.githubusercontent.com/117897120/203368295-78bb8668-c073-4767-b170-1b0b76f46335.png)
![image](https://user-images.githubusercontent.com/117897120/203368321-2d56c13c-cb5c-407e-b0e5-99eb222d6a23.png)

### What version of bitcoin-sv are you using?
List the version number/commit ID, and if it is an official binary, self compiled or a distribution package such as PPA.
1.0.11

### Machine specs:
- OS: Ubuntu 22.04.1 LTS
- CPU: 4 CPUs x Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
- RAM: 24GB
- Disk size: 250GB
- Disk Type (HD/SDD): SSD
- SWAP: 10GB
- DBCACHE: 10GB
- bitcoin-data stored in a 8TB HDD 15rpm NAS mapped with NFS

### Any extra information that might be useful in the debugging process.
This is normally the contents of a `bitcoind.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred.

printtoconsole=1
txindex=1

#ZMQ
zmqpubhashtx=tcp://127.0.0.1:18332
zmqpubhashblock=tcp://127.0.0.1:18332

#Ports - Leave commented for defaults
port=9333
rpcport=9334

#rpc settings
rpcservertimeout=60
rpcworkqueue=600
rpcthreads=16
rpcallowip=0.0.0.0/0 
rpcpassword=password
rpcuser=user

debug=1
dbcache=10000

##MANDATORY (NON-MINING)
excessiveblocksize=10GB
maxstackmemoryusageconsensus=200MB
minminingtxfee=0.00000050

##OPTIONAL (NON-MINING)
maxmempool=25GB
maxsigcachesize=256MiB
maxscriptcachesize=256MiB
maxorphantxsize=64MB

##PERFORMANCE FAQS
blockdownloadtimeoutbaseibdpercent=600
blockdownloadtimeoutbasepercent=200

#multi-threaded options
#threadsperblock=4
#maxparallelblocks
#scriptvalidatormaxbatchsize
#maxparallelblocksperpeer
#maxstdtxvalidationduration=15
#maxcollectedoutpoints=1000000
#maxstdtxnsperthreadratio=10000
#maxnonstdtxvalidationduration

##CONNECTIONS
#maxuploadtarget=10MB
maxconnections=1000

**Comment by pfromberg** [comment]
Hi Juan,

This is not necessarily a problem with the bitcoinsv node. Could you mount the NAS drive via SMB instead of NFS just to check if there is no error in the NFS layer? Also can you check if the NAS server does not have a very old operating system? I know the issues you described occasionally happened in the system layers a few years ago. 

Following issues are unrelated but I want to mention that your hardware is underpowered. A rotary HDD, even though super fast 15krpm, is not ideal, should be solid state. 24GB of RAM is far from enough, I recommend 64GB at least.

**Comment by ZoideNFT** [comment]
Hi pfromberg , 

Same problem using SMB. I've tried to rename bitcoin-data to bitcoin-data_old, created a empty bitcoin-data folder and placed the config file into it. It works, it start the IBD from 0 (obviously) and runs great. With this test, I conclude (maybe) that there is no issue with the storage unless there are not a hidden limit/trigger in the code for.. security reasons.. ¿maybe? don't know..

I know that the hardware is not optimal ;-), but it's enough atm and aparently there is no reason to see this type of error. It must be all great (ok, not superspeed, but great) while it not collapses the consumtion ratios. Disk, RAM, SWAP and NETWORK consumtions are far away from their limits. 

Thanks for your time, I appreciate it.

**Comment by CassClark** [comment]
Internal Note:
this issue has been linked to SVN-2080

**Comment by pfromberg** [comment]
Thanks for those tests Juan. So I close this ticket assuming it is a problem with the used hardware.

**Comment by pfromberg** [comment]
Thanks for the new tests Juan. I am not completely sure what you mean. The problem dissapeared?

**Comment by ZoideNFT** [comment]
No, it not, I tested it with empty folder etc... But when I start it from the original bitcoin-data (the one in wich the node has downloaded 3.5TB blocks) the issue persists.

**Comment by ZoideNFT** [comment]
> 

I'm not compiled the code. I'm using the compiled one.

**Comment by Danconnolly** [comment]
You'll get much better performance if you only move the blocks sub-directory to the NAS and keep the remaining directories locally, on SSD preferably.

**Comment by CassClark** [comment]
hi Juan ( @ZoideNFT )

Please be advised we have produced a tar.gz file for you.  Please use the download link below.   Password to open the file is bsvgithub286.  The download will expire in 14days.

https://share.nchain.network/f/c4c1c13de9984683a14a/?dl=1

**Comment by pfromberg** [comment]
Just to add to the comment from Cass. The build in the download link does not fix any problem but prints more comprehensive information when you get the same disk-full message again. When you run with that build and the same error displays again, can you please post it here? The printout will print the free and available memory separately.

**Comment by ZoideNFT** [comment]
Hi guys, I've tested your build and it shows this:

![image](https://user-images.githubusercontent.com/117897120/204103871-b2e668eb-c841-4cfb-af67-2c4aae34c799.png)

I'm not sure how to understand the data it shows. "additionalbytes"?

thanks!

**Comment by pfromberg** [comment]
Hi Juan. Thanks a lot for investing the time to produce these logs. This is very helpful. We will investigate with priority.

**Comment by CassClark** [comment]
@ZoideNFT 
Hi Juan  
I wanted to check if Patrick had finalised your issue for you before I close this ticket.  I understand he was speaking with you directly. 
thanks

**Comment by ZoideNFT** [comment]
> 

Hi Cass, I've been discussing the issue with Patrick. The point was that its not possible to generate a core dump because the node shuts down properly once it detects the low space error. 

In order to get a core dump, Patrick coded a specific version of the node and I had to compile it. But, I've had no time to execute this compilation because my new daughter born and I had to put this on standby. 

I want to take it back in the next few days, so please excuse me for being missing.

**Comment by CassClark** [comment]
@ZoideNFT 
Hi Juan
Thanks for your reply & congratulations on the birth of your daughter! 

FYI Patrick has since left our company, therefore if you should have any issues with compiling the node, drop me a line here and I'll be happy to help.
